# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/fs/parent/FileSystemManagerParentFactory.cpp
# Commit: 399dce373c59
# Full Hash: 399dce373c5917ac141f7fd341ed7c4dca56144a
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2022-08-31 21:54:47
# Regressor Bug: 1786465
# File Overlap Count: 3
# Description:
#   Bug 1786465 - Move FileSystemDataManager to own source files; r=dom-storage-reviewers,jesup
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D155550
# ==============================================================================

diff -r f62bd47cabd3 -r 399dce373c59 dom/fs/parent/FileSystemManagerParentFactory.cpp
--- a/dom/fs/parent/FileSystemManagerParentFactory.cpp	Wed Aug 31 09:34:53 2022 +0000
+++ b/dom/fs/parent/FileSystemManagerParentFactory.cpp	Wed Aug 31 09:34:54 2022 +0000
@@ -7,6 +7,7 @@
 #include "FileSystemManagerParentFactory.h"
 
 #include "mozilla/StaticPrefs_dom.h"
+#include "mozilla/dom/FileSystemDataManager.h"
 #include "mozilla/dom/FileSystemManagerParent.h"
 #include "mozilla/dom/FileSystemTypes.h"
 #include "mozilla/dom/quota/QuotaCommon.h"
@@ -32,20 +33,6 @@
 
 EntryId GetRootHandle(const Origin& aOrigin) { return "not implemented"_ns; }
 
-// TODO: Replace with real FileSystemDataManager
-class FileSystemDataManager : public FileSystemDataManagerBase {
- public:
-  using result_t = Result<FileSystemDataManager*, nsresult>;
-  static FileSystemDataManager::result_t CreateFileSystemDataManager(
-      const fs::Origin& aOrigin);
-};
-
-FileSystemDataManager::result_t
-FileSystemDataManager::CreateFileSystemDataManager(
-    const fs::Origin& /*aOrigin*/) {
-  return nullptr;
-}
-
 }  // namespace fs::data
 
 mozilla::ipc::IPCResult CreateFileSystemManagerParent(
@@ -67,10 +54,10 @@
 
   // This opens the quota manager, which has to be done on PBackground
   QM_TRY_UNWRAP(
-      fs::data::FileSystemDataManager * dataRaw,
+      fs::data::FileSystemDataManager * dataManagerRaw,
       fs::data::FileSystemDataManager::CreateFileSystemDataManager(origin),
       IPC_OK(), sendBackError);
-  UniquePtr<fs::data::FileSystemDataManager> data(dataRaw);
+  UniquePtr<fs::data::FileSystemDataManager> dataManager(dataManagerRaw);
 
   nsCOMPtr<nsIThread> pbackground = NS_GetCurrentThread();
 
@@ -85,18 +72,18 @@
   // We'll have to thread-hop back to this thread to respond.  We could
   // just have the create be one-way, then send the actual request on the
   // new channel, but that's an extra IPC instead.
-  InvokeAsync(taskqueue, __func__,
-              [origin, parentEp = std::move(aParentEndpoint), aResolver, rootId,
-               data = std::move(data), taskqueue, pbackground]() mutable {
-                RefPtr<FileSystemManagerParent> parent =
-                    new FileSystemManagerParent(taskqueue, rootId);
-                if (!parentEp.Bind(parent)) {
-                  return BoolPromise::CreateAndReject(NS_ERROR_FAILURE,
-                                                      __func__);
-                }
-                // Send response back to pbackground to send to child
-                return BoolPromise::CreateAndResolve(true, __func__);
-              })
+  InvokeAsync(
+      taskqueue, __func__,
+      [origin, parentEp = std::move(aParentEndpoint), aResolver, rootId,
+       dataManager = std::move(dataManager), taskqueue, pbackground]() mutable {
+        RefPtr<FileSystemManagerParent> parent =
+            new FileSystemManagerParent(taskqueue, rootId);
+        if (!parentEp.Bind(parent)) {
+          return BoolPromise::CreateAndReject(NS_ERROR_FAILURE, __func__);
+        }
+        // Send response back to pbackground to send to child
+        return BoolPromise::CreateAndResolve(true, __func__);
+      })
       ->Then(GetCurrentSerialEventTarget(), __func__,
              [aResolver](const BoolPromise::ResolveOrRejectValue& aValue) {
                if (aValue.IsReject()) {